### PR TITLE
task/JM-6638 - Update Notify Table migration scripts

### PR DESCRIPTION
**Pull request by CCHAPMAN01**
### Jira link (if applicable)

https://centralgovernmentcgi.atlassian.net/browse/JM-6638

### Change description ###
- update notify table table migration scripts to:
  -  run inside transactions
  - update the migration log 
  - follow run order naming convention

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines

Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: Postgres old schema to new schema migration/0022 - notify tables.sql [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+truncate table juror_mod.region_notify_template;
+
+with target as (
+ 	insert into juror_mod.region_notify_template (legacy_template_id,message_format,notify_template_id,region_id,region_template_id,template_name,triggered_template_id,welsh_language)
+	select distinct 
+			rnt.legacy_template_id,
+			rnt.message_format,
+			rnt.notify_template_id,
+			rnt.region_id,
```
Comment (ChrisWCGI) [inline Postgres old schema to new schema migration/0022 - notify tables.sql:151]: The Heritage Schema spreadsheet has a note saying region_notify_template.region_id is redundant.
It maps to region_id on court_region but there is no note about whether court_region_redundant.

Is region_notify_template.region_id actually redundant?
Or do we just migrate the anyway?

File: Postgres old schema to new schema migration/0022 - notify tables.sql [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+where 	script_number = '0022b';
+
+do $$
+
+begin
+
+truncate table juror_mod.notify_template_field;
+
+with target as (
+ 	insert into juror_mod.notify_template_field (convert_to_date,database_field,field_length,jd_class_name,jd_class_property,position_from,position_to,template_field,template_id,version)
```
Comment (ChrisWCGI) [inline Postgres old schema to new schema migration/0022 - notify tables.sql:84]: Not migrating notify_template_field.id. According to the New Schema Design spreadsheet ,the ID attribute does exist on the table in JUROR.MOD so presumably we should be migrating it.

File: Postgres old schema to new schema migration/0022 - notify tables.sql (hunk 1 above)
Comment (ChrisWCGI) [reply Postgres old schema to new schema migration/0022 - notify tables.sql:151]: Just checked the staging database and it has region_id on region_notify_template and it also has the court_region table.

I am minded to keep them both in the migration and remove the 'redundant' comment from the schema spreadsheet. Agreed?

Comment (CCHAPMAN01) [reply Postgres old schema to new schema migration/0022 - notify tables.sql:151]: Agreed, the original intention was to largely leave these tables alone so happy to keep the table and FK and update the spreadsheet 👍

**Review by ChrisWCGI — APPROVED** [review]

**Pull request merged**